### PR TITLE
Feature: add tags to selected annotations

### PR DESCRIPTION
**Pull request by bruyeret**
Open a dialog when wlicking "TAG SELECTED" in the action menu of the browse annotation panel.
Remove the embedding of IAnnotation in geojs annotations ("storedAnnotation" option) which caused bugs when updating the IAnnotation.

image

### CODE REVIEW

**Comment by arjunrajlab** [comment]
Just tested, works well, looks good to me for merging!

**Comment by luciemac** [comment]
LGTM

File: src/store/annotation.ts [hunk 1]
```diff
@@ -282,6 +282,41 @@ export class Annotations extends VuexModule {
     this.setSelected([]);
   }
 
+  @Action
+  public async tagAnnotationIds({annotationIds, tags}: {annotationIds: string[], tags: string[]}) {
+    sync.setSaving(true);
+    await Promise.all(
+      annotationIds
+        .map(annotationId => this.annotations.findIndex((annotation: IAnnotation) => annotation.id === annotationId))
```
Comment (luciemac) [inline src/store/annotation.ts:290]: Don't you think it could be useful to have a helper method to get annotations from annotationsIds ?

Comment (bruyeret) [reply src/store/annotation.ts:290]: In this case I need the index to use it later but that's something that could be helpful in other places in the code
But I think it should be done in another pull request with eventually other reformatting

File: src/components/AnnotationViewer.vue [hunk 2]
```diff
@@ -334,12 +334,19 @@ export default class AnnotationViewer extends Vue {
...
+    this.textLayer.features().forEach((feature: any) => {
+      this.textLayer.deleteFeature(feature);
+    });
 
     if (this.showTooltips) {
-      const annotationsToTooltip = this.annotationLayer
+      const displayedAnnotations: IAnnotation[] = this.annotationLayer
         .annotations()
-        .map((annotation: any) => annotation.options("storedAnnotation"))
+        .map((geojsAnnotation: any) => {
```
[the remark this answers is not in the document]
Comment (bruyeret) [reply src/components/AnnotationViewer.vue:344]: It is a [geojs annotation](https://opengeoscience.github.io/geojs/apidocs/geo.annotation.html) but I didn't find a ts type for it. This `.map()` finds the corresponding IAnnotation in the store and the filter after it receives a `IAnnotation | undefined`.

**Pull request merged**